week3/cardgame/src/App.jsx

```diff
@@ -0,0 +1,47 @@
+import { Global, ThemeProvider } from '@emotion/react';
+import { useState, useEffect } from "react";
+import globalStyle from "./styles/globalStyle";
+import theme from "./styles/theme";
+import styled from "@emotion/styled";
+import Header from "./Header"
+import CardTable from "./CardTable";
+import Modal from "./Modal";
+
+
+
+function App() {
+  const [score, setScore] = useState(0);
+  const [showModal, setShowModal] = useState(false);
+
+  useEffect(() => {
+    if (score === 5) {
+      setShowModal(true);
+    }
+  }, [score]);
+
+  const handleCloseModal = () => {
+    setShowModal(false);
+    window.location.reload()
+  };
+  
+  return (
+    <>
+    <Global styles={globalStyle} />
+    <ThemeProvider theme = {theme}>
+      <ContentWrapper>
+        <Header score = {score}/>
+        <CardTable setScore = {setScore} />
```

Comment:
p3: 너무 잘했는데! 여기서 setScore와 같이 자식 컴포넌트의 상태를 바꾸는 setState를 props로 넘기는 건 고민해볼 필요가 있을 것 같아! 
내 생각에는 보통은 같은 역할을 하더라도 간단한 함수를 만들어서 처리하는 방식이 재사용의 측면에서도, 다른 추가적인 요구사항이 생겨도 유지 보수의 측면에도 좋을 것 같아! 
이거와 관련해서는 링크를 하나 남길게! 한 번 보고 고민해봐~!


```suggestion
    const matchScore = () => {
        setScore((prev) => prev+1)
    }
        <CardTable matchScore={matchScore}  />
```

https://velog.io/@liswktjs/%EC%BB%B4%ED%8F%AC%EB%84%8C%ED%8A%B8-%EB%B6%84%EB%A6%AC%ED%95%A0-%EB%95%8C-%EC%96%B4%EB%96%BB%EA%B2%8C-%ED%95%98%EC%8B%9C%EB%82%98%EC%9A%94

Reply:
오.. setState를 넘기는 것에 대해 아무런 생각이 없었는데, 도영님 리뷰랑 아티클을 보고나니 setState 자체를 넘기는 것보단 함수를 하나 만들어서 처리하는 방식이 훨씬 좋은 것 같네요!! 감사합니다! 반영했습니다 👍🏻